src/components/input/input.tsx

```diff
@@ -833,22 +833,21 @@ export class Input
   }
 
   syncHiddenFormInput(input: HTMLInputElement): void {
-    if (this.type === "number") {
-      input.type = "number";
+    const { type } = this;
+
+    input.type = type;
+
+    if (type === "number") {
       input.min = this.min?.toString(10) ?? "";
       input.max = this.max?.toString(10) ?? "";
```

Comment:
Nitpick: I know you didn't touch this, but `10` is the default radix and I think putting it there is more confusing than omitting it.

Reply:
We can remove it [since it's the default](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toString#radix). Maybe it was conflated with [`parseInt` radix recommendation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#radix): 

> If 0 or not provided, the radix will be inferred based on string's value. Be careful — this does NOT always default to 10!